source/Irrlicht/CBawFile.h

```diff
@@ -107,6 +144,15 @@ namespace irr { namespace core
 		uint64_t attrBufPtrs[scene::EVAI_COUNT];
 		uint64_t idxBufPtr;
 	} PACK_STRUCT;
+
+	struct FinalBoneHierarchyBlobV1 : Blob
+	{
+		FinalBoneHierarchyBlobV1(size_t _bCnt, size_t _numLvls, size_t _kfCnt);
```

Comment:
Maybe have a constructor/static create function directly from the object being saved

+see my note about variable size blobs

Reply:
>Maybe have a constructor/static create function directly from the object being saved
  
We already talked about this and decided to load blob buffer with variable-size data outside ctor since it has to rely on the fact that the memory actually is allocated (and check it from inside ctor) and it's kinda dangerous

Reply:
True, thats why I'm pondering making the constructor private and instead making a static "createOverMemory" function which will call the constructor through placement new in private and check the memory size provided...

It would have to be a static function and a private constructor as constructors are not allowed to not create the object or to return anything else than the object.